synthesizer/process/src/lib.rs

```diff
@@ -202,11 +202,26 @@ impl<N: Network> Process<N> {
 
     /// Returns the stack for the given program ID.
     #[inline]
-    pub fn get_stack(&self, program_id: impl TryInto<ProgramID<N>>) -> Result<&Stack<N>> {
+    pub fn get_stack(&self, program_id: impl TryInto<ProgramID<N>>) -> Result<Arc<Stack<N>>> {
```

Comment:
I think this could be reimplemented as the following one-liner:
```
Ok(Arc::clone(self.get_stack_ref(program_id)?))
```
(as long as `get_stack_ref` can return a `Result<&Arc<Stack<N>>>` instead of `Result<&Stack<N>>`, which should be fine)

Reply:
Neat. Let me know what you think: a3f93f00a

Note, the reason I introduced the `_ref()` functions was to avoid littering the codebase with `as_ref()`'s or refactoring a bunch of function signatures to use `Arc`'s instead of refs (which I looked into again but is very involved).

Reply:
that makes sense, and I think the end result is a good compromise; neat APIs are nice, but code duplication can end up resulting in future errors, so I try to avoid it whenever possible